### PR TITLE
Add config to Screenshot plugin for Screenshot clan member deaths

### PR DESCRIPTION
**Pull request by pwatts6060**
Feature request from discord implemented: https://discord.com/channels/301497432909414422/301497432909414422/878905125878501406

is there any way we could get an option to screenshot clan members deaths? we have our own plus friends, but with the new clan system I think it would be useful

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 1]
```diff
@@ -306,6 +306,10 @@ else if (player != client.getLocalPlayer() && (player.isFriendsChatMember() || p
 			{
 				takeScreenshot("Death " + player.getName(), SD_DEATHS);
 			}
+			else if (player != client.getLocalPlayer() && player.isClanMember() && config.screenshotClanDeath() && player.getCanvasTilePoly() != null)
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java:309]: This can be merged into the above statement.

**Review by raiyni — CHANGES_REQUESTED** [review]

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java (hunk 1 above)
Comment (pwatts6060) [reply runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: Should be good now, nice spot

Comment (pwatts6060) [reply runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: @raiyni ok finally got around to fixing this

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 2]
```diff
@@ -302,7 +302,8 @@ public void onActorDeath(ActorDeath actorDeath)
 			{
 				takeScreenshot("Death", SD_DEATHS);
 			}
-			else if (player != client.getLocalPlayer() && (player.isFriendsChatMember() || player.isFriend()) && config.screenshotFriendDeath() && player.getCanvasTilePoly() != null)
+			else if (player != client.getLocalPlayer() && (player.isFriendsChatMember() || player.isFriend()) && config.screenshotFriendDeath() && player.getCanvasTilePoly() != null
+				|| player != client.getLocalPlayer() && player.isClanMember() && config.screenshotClanDeath() && player.getCanvasTilePoly() != null)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java:306]: The local player and canvas tile poly checks can be pulled out as they're common to both conditions. I'd recommend writing the check like this:
```suggestion
			else if (player != client.getLocalPlayer()
				&& player.getCanvasTilePoly() != null
				&& (((player.isFriendsChatMember() || player.isFriend()) && config.screenshotFriendDeath())
					|| (player.isClanMember() && config.screenshotClanDeath())))
```

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotConfig.java [hunk 3]
```diff
@@ -181,11 +181,23 @@ default boolean screenshotFriendDeath()
 		return false;
 	}
 
+	@ConfigItem(
+		keyName = "clanDeath",
+		name = "Screenshot Clan Deaths",
+		description = "Configures whether or not screenshots are automatically taken when clan chat members die.",
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotConfig.java:187]:
```suggestion
		description = "Configures whether or not screenshots are automatically taken when clan members die.",
```

**Review by Nightfirecat — CHANGES_REQUESTED** [review]

**Comment by pwatts6060** [comment]
Ok should be good now thanks

**Pull request merged**